### DISCUSSION TITLE
protobuf symbol duplicate

### DISCUSSION THREAD

**Discussion by gyujin-kim** [microsoft/onnxruntime#17248] [Other Q&A] [answered]
I am using protobuf framework and onnxruntime-c framework simultaneously in ios.
But when I build the app I get a symbol duplicate warning and
When I run the app, protobuf is crashing.
Is protobuf included in onnxruntime?
Is there any workaround?

onnxruntime-c : 1.15.1
protobuf : 3.9 (3.21.12 has the same issue)

**Comment by snnn** [comment]
Yes, it is included. 
I don't have experience with iOS, but on Linux they are hided and won't cause problems

**Comment by gyujin-kim** [comment]
@snnn
thanks for reply.

Acctually, i have using protobuf already by dependency.
And this problem has occured to attach new solution using onnxruntime.
In this case, symbol duplication is occured and crash when it run.
It is both issued the iPhone12 and M1 Mac.
Does any  method to hide internal symbol in onnxruntime?
And any suggestions?

* dependency in Podfile
s.dependency 'Protobuf', '~> 3.21.12' (version is updated from 3.9)
s.dependency 'onnxruntime-c', '1.15.1' (added for new solution)

**Reply by snnn** [reply]
On Linux or Windows you can have multiple protobuf libraries in the same process, they won't have conflicts. Maybe macOS also has a similar mechanism.  See: https://gcc.gnu.org/wiki/Visibility

**Reply by gyujin-kim** [reply, marked as answer]
Hi @snnn
i have some news about this issue.

I was able to get through this problem in two ways.
i guess that the released ios framework is built with static framework.
(build script default)

I was able to get through this in two ways.
First, we were able to use the static protobuf library by adding a new target to our app.
In my case it is not suitable because onnxruntime is used by native library.
Another way is to rebuild onnruntime with the dynamic framework and then link to it using 'vendored_framework' in your podspec.
My choice is the second and it solved my problem.
thanks.

**Reply by snnn** [reply]
Thanks for sharing!
